hangman/hangman.c

```diff
@@ -1,20 +1,182 @@
-	"             |         |         |   O     |   O     |   O     |   O     |   O     |   O  ",
-	"             |         |         |         |   +     | --+     | --+--   | --+--   | --+--",
-	"             |         |         |         |   |     |   |     |   |     |   |     |   |  ",
-	"             |         |         |         |         |         |         |  /      |  / \\ ",
-	"             |         |         |         |         |         |         |         |      ",
-	"/*****\\   /*****\\   /*****\\   /*****\\   /*****\\   /*****\\   /*****\\   /*****\\   /*****\\   "
+	"         +        +----    +----    +----    +----    +----    +----    +----  ",
+	"         |        |        |   O    |   O    |   O    |   O    |   O    |   O  ",
+	"         |        |        |        |   +    | --+    | --+--  | --+--  | --+--",
+	"         |        |        |        |   |    |   |    |   |    |   |    |   |  ",
+	"         |        |        |        |        |        |        |  /     |  / \\ ",
+	"         |        |        |        |        |        |        |        |       ",
+	"/*****\\  /*****\\  /*****\\  /*****\\  /*****\\  /*****\\  /*****\\  /*****\\  /*****\\"
 };
 
+// Display Hang States
+void display_stage(int stage) {
+	int start = 9 * stage;
+	int end = start + 9;
+	printf("\n");
+	for(int i = 0; i < 7; i++) {
+		for (int j = start; j < end; j++) {
+			printf("%c", HANG_STAGES[i][j]);
+		}
+		printf("\n");
+	}
+}
+
+
 int main(int argc, char *argv[]) {
   /* Your code here */
-  
+	if (argv[1] != NULL) {
+	char *word = argv[1];
+	char guess[40];
+	char guesses[27]; // Guesses only needs space for each character in the alphabet.
+	guesses[0] = '\0';
+
+	int position = 0;
+	int stage = 0;
+	int match = 0;
+	int guess_length = 0;
+	int word_length = 0;
+	int guesses_count = 0;
+	int correct_guesses = 0;
+
+	for (int i = 0 ; word[i]; i++) { 
+		word[i] = tolower(word[i]);
+		word_length++;
+	}
+
+	// Set Blanks
+	char blanks[word_length];
+	for (int i = 0; i < word_length; i++) {
+			blanks[i] = '_';
+			blanks[i + 1] = '\0';
+	}
+	
+	// Main Loop
+	while (free) {
+
+		// Display Hanging Stage
+		display_stage(stage);
+
+		// Display Blanks
+		printf("\nWord: ");
+		for (int i = 0; i < word_length; i++) {
+			printf("%c ", blanks[i]);
+		}
+
+		printf("\n");
+
+		// Display Guessed Letters
+		printf("\nGuesses: ");
+		for (int i = 0; i < guesses_count; i++) {
+			printf("%c ", guesses[i]);
+		}
+
+		printf("\n\n");
+
+		// Loss Handling
+		if (stage == 8) {
+			printf("\n----\nHANGED!\n----\n");
+			break;
+		}
+
+		// Win Handling
+		if (correct_guesses == word_length) {
+			printf("\n----\nCORRECT! YOU LIVE ANOTHER DAY!\n----\n");
+			break;
+		}
+		
+		// User Input
+		printf("Enter your guess. You may guess one character, or the whole word (enter _q to quit):\n");
+		scanf("%s", guess);
+
+		for (int i = 0; guess[i]; i++) {
+			guess[i] = tolower(guess[i]);
+			guess_length++;
+		}
+
```

Comment:
if there was a way to optimize your solution, what would you do differently to eliminate the nested loops and extra conditionals?